tests/unit-tests/platforms/mesa/kms-utils/test_drm_mode_resources.cpp

```diff
@@ -143,7 +143,12 @@ class FakeDRMObjectsWithProperties
         memset(prop.get(), 0, sizeof(*prop));
 
         prop->prop_id = id;
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpragmas" // Don't worry if g++ doesn't recognise the next
+#pragma GCC diagnostic ignored "-Wstringop-truncation"
         strncpy(prop->name, property.c_str(), sizeof(prop->name));
```

Comment:
This call is potentially wrong, correct? ie: gcc is *correctly* warning that `prop->name` might not be null-terminated after this call.

Rather than disable the warning, shouldn't we be passing `sizeof(prop->name) - 1` as the size to `strncpy` and then explicitly setting `prop->name[sizeof(prop->name) - 1] = '\0'` instead?

Reply:
I was reading drmModePropertyRes::name as a nul padded field of DRM_PROP_NAME_LEN characters. (In which case the call is correct.)

But there is precedence for your reading. I see:

       r->name[DRM_PROP_NAME_LEN-1] = 0;

https://sourcecodebrowser.com/libdrm/2.4.5/xf86drm_mode_8h.html#aee000c44bad980e06a89016eaf45bf4d